sph/include/sph/find_neighbors.hpp

```diff
@@ -28,7 +28,9 @@ void findNeighborsSph(const T* x, const T* y, const T* z, T* h, LocalIndex first
         int iteration = 0;
         while (ngmin > nc[i] || nc[i] > ngmax && iteration++ < maxIteration)
         {
-            h[id] = updateH(ng0, nc[i], h[id]);
+            if (nc[i] == 0)
+                h[id] *= 1.25;
+            else { h[id] = updateH(ng0, nc[i], h[id]); }
```

Comment:
`nc[i]` does not include the self reference. I think here it would be permissible to use `nc[i] + 1`  in `updateH` and avoid the division by zero problem. This would also address it on the GPU while this "fix" here would only be used on the GPU.

Reply:
Addressed here
https://github.com/unibas-dmi-hpc/SPH-EXA/pull/406